geth/common/profiling.go

```diff
@@ -0,0 +1,41 @@
+package common
+
+import (
+	"os"
+	"path/filepath"
+	"runtime/pprof"
+)
+
+const (
+	PROFILING_FILENAME = "status.prof"
+)
+
+type Profiling struct {
+	file *os.File
+}
+
+// Returns new profiling struct
+func NewProfiling() *Profiling {
+	return &Profiling{}
+}
+
+// Enables CPU profiling for the current process.
+// While profiling, the profile will be buffered and written to file in folder dataDir
+func (p *Profiling) Start(dataDir string) *Profiling {
+	if p.file == nil {
```

Comment:
Well. Can we keep it more simple? Without the `Profiling` struct, just as a wrapper to StartCPUProfile and StartHeadProfile.